cargo-near/src/lib.rs

```diff
@@ -25,31 +30,48 @@ pub enum NearCommand {
     /// Generates ABI for the contract
     #[clap(name = "abi")]
     Abi(AbiCommand),
+    /// Build a NEAR contract and optionally embed ABI
+    #[clap(name = "build")]
+    Build(BuildCommand),
 }
 
 #[derive(Debug, clap::Args)]
-#[clap(name = "abi")]
+#[clap(setting = AppSettings::DeriveDisplayOrder)]
 pub struct AbiCommand {
+    /// Include rustdocs in the ABI file
+    #[clap(long)]
+    pub doc: bool,
     /// Path to the `Cargo.toml` of the contract to build
-    #[clap(long, parse(from_os_str))]
+    #[clap(long, parse(from_os_str), value_name = "PATH")]
     pub manifest_path: Option<PathBuf>,
-    /// Include rustdocs in the ABI file
-    #[clap(long, takes_value = false)]
+}
+
+#[derive(Debug, clap::Args)]
+#[clap(setting = AppSettings::DeriveDisplayOrder)]
+pub struct BuildCommand {
+    /// Build contract in release mode, with optimizations
+    #[clap(short, long)]
+    pub release: bool,
+    /// Embed the ABI in the contract binary
+    #[clap(long)]
+    pub embed_abi: bool,
+    /// Include rustdocs in the embedded ABI
+    #[clap(long, requires = "embed-abi")]
     pub doc: bool,
+    /// Do not generate ABI for the contract
+    #[clap(long, conflicts_with_all = &["doc", "embed-abi"])]
+    pub no_abi: bool,
+    /// Copy final artifacts to the this directory
+    #[clap(long, parse(from_os_str), value_name = "PATH")]
+    pub out_dir: Option<PathBuf>,
+    /// Path to the `Cargo.toml` of the contract to build
+    #[clap(long, parse(from_os_str), value_name = "PATH")]
+    pub manifest_path: Option<PathBuf>,
```

Comment:
I still feel like this should at least have default value set to `Cargo.toml` instead of unwrapping it in the middle of the code flow. This also has the benefit of showing up in the help command.

```suggestion
    #[clap(
        long,
        parse(from_os_str),
        default_value = "Cargo.toml",
        value_name = "PATH"
    )]
    pub manifest_path: PathBuf,
```

Reply:
> This also has the benefit of showing up in the help command.

The description already hints at `Cargo.toml`.

```console
        --manifest-path <PATH>    Path to the `Cargo.toml` of the contract to build
```

This way, it's more consistent with `cargo build`:

```console
        --manifest-path <PATH>      Path to Cargo.toml
```

`cargo build`'s behavior without specifying a manifest path is to scan parent directories up to the root for a `Cargo.toml` and print this when it can't find one:

```console
error: could not find `Cargo.toml` in `/some/path/somewhere` or any parent directory
```

`cargo near`'s behavior right now is to check the current directory for a `Cargo.toml` and print this when it can't find one:

```console
error: manifest path `Cargo.toml` does not exist
```